### DISCUSSION TITLE
Simpler Signin and connection

### DISCUSSION THREAD

**Discussion by TechyChap** [abuzuhri/Amazon-SP-API-CSharp#667] [General]
Starting October 2, 2023, SP-API no longer requires the use of [AWS Identity and Access Management](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) (IAM) or [AWS Signature Version 4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html), which means developers will not need to register for IAM resources or sign SP-API requests with AWS Signature Version 4.
SP-API will continue to use [Login with Amazon](https://developer.amazon.com/docs/login-with-amazon/documentation-overview.html) (LWA) access tokens for each [SP-API request](https://developer-docs.amazon.com/sp-api/docs/connecting-to-the-selling-partner-api). LWA is based on OAuth 2.0, which has been broadly adopted for user-authorized exchanges across sites.

**Which marketplaces are affected?**
This change applies to all regions.
**Who is affected?**
All developers who registered with SP-API before October 2, 2023.
**What action is required?**
No immediate action is required. These changes are integrated into our production environment. For requests with AWS Signature Version 4, we’ll disregard the signature and proceed with LWA authorization.
MWS developers
MWS developers who are actively migrating to SP-API can use the following guidance:
•	If you haven’t set up AWS IAM resources, you will not have to complete those steps.
•	If you completed the IAM resource setup, but haven’t used them, these resources are no longer required.
•	If you are currently migrating, you can choose to either continue to sign your SP-API requests with AWS Signature Version 4, or stop using at your convenience.
•	Refer to the [Migration Hub](https://developer-docs.amazon.com/sp-api/page/migration-hub) for more details.
**Developers using SDKs**
Version 2.0 of the Java and C# Software Development Kits are available starting October 2, 2023. No immediate action is required, but we highly recommend using the updated SDKs as they save time during signing and provide more details for LWA Authorization exceptions, including information on errors returned during LWA Authorization using the new custom LWAException.
SDK versions 1.0 for [Java](https://protect-eu.mimecast.com/s/cq25CD8R8Sj8KnQuk9Sqi?domain=sellercentral-europe.amazon.com) and [C#](https://protect-eu.mimecast.com/s/oxRrCExlxi1ZJgKSBKK2Y?domain=sellercentral-europe.amazon.com), with optional AWS signing, are still available, but will be deprecated by the end of 2023.
For more information on connecting to SP-APIs using SDKs, refer to [Automate your SP-API calls using Java SDK](https://developer-docs.amazon.com/sp-api-blog/docs/automate-your-sp-api-calls-using-java-sdk) and [Automate your SP-API calls using C# SDK](https://developer-docs.amazon.com/sp-api-blog/docs/automate-your-sp-api-calls-using-the-c-sdk).
**Resources**
If you have questions or need assistance, contact the Selling Partner API Developer Support team via the [Contact Us](https://developer.amazonservices.com/support) form.

**Comment by J-W-Chan** [comment]
Just comment this line of code
[https://github.com/abuzuhri/Amazon-SP-API-CSharp/blob/9eadcff5dfba5520eabf916f6698f5184a4dd53f/Source/FikaAmazonAPI/Services/RequestService.cs#L121C16-L121C16](https://github.com/abuzuhri/Amazon-SP-API-CSharp/blob/9eadcff5dfba5520eabf916f6698f5184a4dd53f/Source/FikaAmazonAPI/Services/RequestService.cs#L121C16-L121C16)

**Comment by TechyChap** [comment]
Looks like this is resolved in [v1.7.8](https://github.com/abuzuhri/Amazon-SP-API-CSharp/commit/fd741bc21f41e121a922291c613651ca864c5379)

**Comment by abuzuhri** [comment]
Yes I do that in last version
